src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksVendorDaoJdbc.java

```diff
@@ -10,79 +13,132 @@
+        ParameterizedSqlFactory sqlFactory = new ParameterizedSqlFactory("update kfs.cu_pmw_vendor_t set");
+
         if (StringUtils.isNotBlank(pmwRequestStatus)) {
-            sql.append(" pmw_req_stat = '" + pmwRequestStatus + "',");
+            sqlFactory.appendSql(" pmw_req_stat = ?,", pmwRequestStatus);
         }
-        sql.append(" kfs_vnd_proc_stat = '" + kfsVendorProcessingStatus + "',");
-        sql.append(" proc_ts = '" + PROCESSING_TIMESTAMP_SQL_FORMATTER.format(processingTimeStamp) + "'");
+
+        sqlFactory.appendSql(" kfs_vnd_proc_stat = ?,", kfsVendorProcessingStatus);
+        sqlFactory.appendSql(" proc_ts = ?", PROCESSING_TIMESTAMP_SQL_FORMATTER.format(processingTimeStamp));
+
         if (StringUtils.isNotBlank(kfsAchProcessingStatus)) {
-            sql.append(", kfs_ach_proc_stat = '" + kfsAchProcessingStatus + "'");
+            sqlFactory.appendSql(", kfs_ach_proc_stat = ?", kfsAchProcessingStatus);
         }
         if (StringUtils.isNotBlank(supplierUploadStatus)) {
-            sql.append(", supp_upld_stat = '" + supplierUploadStatus + "'");
+            sqlFactory.appendSql(", supp_upld_stat = ?", supplierUploadStatus);
         }
         if (StringUtils.isNotBlank(kfsVendorDocumentNumber)) {
-            sql.append(", pven_fdoc_nbr = '" + kfsVendorDocumentNumber + "'");
+            sqlFactory.appendSql(", pven_fdoc_nbr = ?", kfsVendorDocumentNumber);
         }
         if (ObjectUtils.isNotNull(vendorHeaderGeneratedIdentifier)) {
-            sql.append(", vndr_hdr_gnrtd_id = '" + vendorHeaderGeneratedIdentifier.intValue() + "'");
+            sqlFactory.appendSql(", vndr_hdr_gnrtd_id = ?", vendorHeaderGeneratedIdentifier);
         }
         if (ObjectUtils.isNotNull(vendorDetailAssignedIdentifier)) {
-            sql.append(", vndr_dtl_asnd_id = '" + vendorDetailAssignedIdentifier.intValue() + "'");
+            sqlFactory.appendSql(", vndr_dtl_asnd_id = ?", vendorDetailAssignedIdentifier);
         }
         if (ObjectUtils.isNotNull(kfsAchDocumentNumber)) {
-            sql.append(", paat_fdoc_nbr = '" + kfsAchDocumentNumber + "'");
+            sqlFactory.appendSql(", paat_fdoc_nbr = ?", kfsAchDocumentNumber);
         }
-        sql.append(" where id = '" + id + "'");
-        
-        return sql.toString();
+        sqlFactory.appendSql(" where id = ?", id);
+
+        return sqlFactory.getPreparedSqlStatement();
     }
 
+    private class ParameterizedSqlFactory {
+        private StringBuilder sqlStringBuilder;
+        private ArrayList parameters;
+
+        public ParameterizedSqlFactory(String initialSql) {
+            sqlStringBuilder = new StringBuilder(initialSql);
+            parameters = new ArrayList();
```

Comment:
Initialization of the list should use generics. For this case, you can specify an empty generic-type placeholder and the compiler should be able to auto-detect what type should be used.